core-libs/setup/ssr/error-handling/server-error-response-factory/cms-page-not-found-server-error-response-factory.ts

```diff
@@ -24,30 +24,15 @@ export class CmsPageNotFoundServerErrorResponseFactory
   implements ServerErrorResponseFactory
 {
   protected config = inject(OccConfig);
+  protected injector = inject(Injector);
 
   hasMatch(error: unknown): boolean {
-    const occConfig = this.config.backend?.occ;
-    if (occConfig) {
-      const { baseUrl, prefix, endpoints } = occConfig;
-      const pagesEndpoint = endpoints?.pages;
-      const endpoint =
-        pagesEndpoint &&
-        (typeof pagesEndpoint === 'string'
-          ? pagesEndpoint
-          : pagesEndpoint.default);
-      // The OccEndpointsService can't be used due to circular dependency injection
-      // Optional BaseSiteService depends on ngrx, which has dependency on the ErrorHandler
-      const expectedUrlRegex = this.buildUrlRegex({
-        baseUrl,
-        prefix,
-        endpoint,
-      });
-      return (
-        error instanceof HttpErrorResponse &&
-        expectedUrlRegex.test(error.url ?? '')
-      );
-    }
-    return false;
+    const occEndpointsService = this.injector.get(OccEndpointsService);
```

Comment:
Let's add a code comment, e.g.
```suggestion
    // OccEndpointsService could not be injected in the constructor, due to a circular dependency:
    //   ErrorHandler -> *CmsPageNotFoundServerErrorResponseFactory* -> OccEndpointsService -> 
    //   -> BaseSiteService -> NgRx -> ErrorHandler
    const occEndpointsService = this.injector.get(OccEndpointsService);
```